scipy/stats/_qmc.py

```diff
@@ -306,6 +308,39 @@ def discrepancy(
                          f" {set(methods)!r}")
 
 
+def geometric_discrepancy(
+        sample: npt.ArrayLike,
+        method: Literal["mindist", "mst"] = "mindist",
+        metric: str = "euclidean") -> float:
+    """Discrepancy of a given sample based on geometric metrics.
+
+    Parameters
+    ----------
+    sample : array_like (n, d)
+        The sample to compute the discrepancy from.
+    method: str, optional
+        The method to use. One of ``mindist`` for minimum distance or ``mst`` for minimum spanning tree.
+    metric: str or function, optional
+        The distance metric to use. See the documentation for ``scipy.spatial.distance.pdist``.
+
+    Returns
+    -------
+    discrepancy : float
+        Discrepancy.
+    """
+    distances = distance.pdist(sample, metric=metric)
+    if method == "mindist":
+        return np.min(distances[distances.nonzero()])
+    elif method == "mst":
+        fully_connected_graph = distance.squareform(distances)
+        mst = minimum_spanning_tree(fully_connected_graph)
+        distances = mst[mst.nonzero()]
+        # TODO consider returning the standard deviation or a tuple
+        return np.mean(distances)
+    else:
+        raise ValueError(f"{method!r} is not a valid method. It must be one of {{mindist, mst}}")
```

Comment:
We can also add an input validation for `sample`. We could allow to pass something else than a unit hypercube though. Also to question as we could want to have consistency with the other metric.

Reply:
Done, I've added the sample validations as used in `discrepancy`, including the check that the sample is within a unit hypercube.